### PR TITLE
[BEAM-4385][SQL] Add LIKE operator to Beam SQL

### PR DESCRIPTION
**Pull request by amaliujia**
Add ```LIKE``` operator to Beam SQL to support the use case like ```SELECT * FROM table WHERE str_col LIKE <regex>```.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

to: @kennknowles could you please take a look at this PR?

### CODE REVIEW

**Comment by kennknowles** [comment]
Sorry for the delay!

R: @apilloud 

I think you might be the best choice for a review pass.

File: sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/interpreter/operator/comparison/BeamSqlLikeExpression.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  public BeamSqlLikeExpression(List<BeamSqlExpression> operands) {
+    super(operands);
+  }
+
+  @Override
+  public Boolean compare(CharSequence leftValue, CharSequence rightValue) {
+    String regexStr = rightValue.toString();
+
+    return Pattern.compile(regexStr).matcher(leftValue).matches();
```
Comment (apilloud) [inline sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/interpreter/operator/comparison/BeamSqlLikeExpression.java]: This should be using `SqlFunctions.like()`

**Comment by amaliujia** [comment]
@apilloud I addressed your comments. Can you take another look?

File: sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/interpreter/operator/comparison/BeamSqlLikeExpression.java (hunk 1 above)
Comment (amaliujia) [reply sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/interpreter/operator/comparison/BeamSqlLikeExpression.java]: Fixed.

**Comment by apilloud** [comment]
LGTM.

**Comment by amaliujia** [comment]
@kennknowles for final approval and merging.

**Comment by kennknowles** [comment]
Great. Taking a peek!

**Pull request merged**